### PR TITLE
[GEN-1109] Setup base mkdocs for genie

### PR DESCRIPTION
**Pull request by rxu17**
**Purpose:** This draft PR adds an auto-documentation tool `mkdocs` and `mkdocstrings` for the Genie repo, setting up the initial skeleton and base pages for the genie docs. It helps auto-document our codebase as well as allowing us to add custom markdown pages. 

This PR is just meant to set up the skeleton of the docs so we can get started with documenting our codebase and looking into areas of lacking documentation / improvement, refining specific things like font size, headings, etc will be done in a separate ticket.

**Limitations:** We will not be able to auto-document our R scripts as there isn't an extension similar to `mkdocstrings` for R. We can use a mix of `roxygen2` and `knitr` to convert our R code to Rmarkdown files -> markdown files which can then be rendered in mkdocs. However this way is more hacky. We plan to convert all our R scripts to Python eventually / move the dashboard R scripts to Nextflow module so this may not be a concern.

**Changes:**
Initial vision and structure for the genie docs:
```
/project-root
├── Home
├── Getting Started
├── Tutorials
├── Reference (this page contains all the technical references to our pipeline code (all the modules and functions)
│   ├── Pipeline Commands (contains the scripts that have the command args to run the pipeline steps)
│   ├── Main Pipeline Steps
│    │    ├── Input to Database (include all scripts unique to validation / processing like input_to_database, validation, process_mutation, write_invalid_reasons)
│    │    ├── Database to Staging (include all scripts unique to consortium release like database_to_staging.py)
│    │    ├── Consortium To Public (include all scripts unique to public release like consortium_to_public.py)
│   ├── Fileformats (all of our fileformats including base fileformat class)
│   ├── Helper Modules (all of the helper tools used across the pipeline - extract, load, transform, process_functions)

```
**Testing:**
In the root directory run `mkdocs serve` to view the live docs

**Examples:**
image

image

image

### CODE REVIEW

**Comment by thomasyu888** [comment]
Thanks @rxu17 for your work here!  I agree that we don't have to worry about auto generating the R code documentation for now.  It'd be great if this was linked to github pages!

**Comment by rxu17** [comment]
You can see the live site at: https://sage-bionetworks.github.io/Genie/

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# AACR Project GENIE
+
+[![PyPi](https://img.shields.io/pypi/v/aacrgenie.svg?style=for-the-badge&label=PyPi&logo=PyPi)](https://pypi.org/project/aacrgenie)
+[![Docker Automated](https://img.shields.io/docker/automated/sagebionetworks/genie.svg?style=for-the-badge&logo=docker)](https://hub.docker.com/r/sagebionetworks/genie)
+[![GitHub CI](https://img.shields.io/github/actions/workflow/status/Sage-Bionetworks/Genie/ci.yml?branch=develop&style=for-the-badge&logo=github)](https://github.com/Sage-Bionetworks/Genie)
+
+
+## Introduction
+
+This repository documents code used to gather, QC, standardize, and analyze data uploaded by institutes participating in AACR's Project GENIE (Genomics, Evidence, Neoplasia, Information, Exchange).
```
Comment (thomasyu888) [inline docs/index.md]: Lets link to the portal URL?

Comment (rxu17) [reply docs/index.md]: Like move portal URL to here, and put the GH pages site to here?
image

Comment (thomasyu888) [reply docs/index.md]: @rxu17 hmmm, i think its valuable to have the portal URL be linked in as many places as possible.  I did think about putting out documentation URL there.  I go back and forth, but I suppose people who land on the GENIE github probably want documentation and not the portal.

Thoughts @Chelsea-Na ?

Comment (rxu17) [reply docs/index.md]: Got it, I added it to that intro page

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!

**Comment by rxu17** [comment]
Per the weekly main genie check-in today, Chelsea also gave this the green light

**Pull request merged**